### PR TITLE
feat: upgrade to angular6 and rxjs6

### LINKED ISSUE #161: Please add support for Angular 6.0
Angular 6.0 uses RxJS 6.x

Now `angular2-toaster 5.0.1` fails on compile:
```./node_modules/angular2-toaster/src/toaster.service.js
Module not found: Error: Can't resolve 'rxjs/Observable' in '/private/tmp/st-cloud-six/node_modules/angular2-toaster/src'
```

I can use `rxjs-compat` module, but it is workaround

### PR DESCRIPTION
**Pull request by sguiheux**
close #161

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,7 +84,8 @@
     "remap-istanbul": "0.10.1",
     "rimraf": "2.5.4",
     "rollup": "0.55.3",
-    "rxjs": "^5.4.2",
+    "rxjs": "^6.1.0",
+    "rxjs-compat": "^6.1.0",
```
Comment (razfriman) [inline package.json]: Can we remove this dependency since you updated the the code to use the new RxJS 6.X.X format?

**Comment by sguiheux** [comment]
Done

**Comment by Stabzs** [comment]
@sguiheux thanks for the work on this! 

I'd like to test and merge in this PR, but it is currently failing the Travis CI build.  As a result, until the build is passing, I'll look at making the changes locally myself and I'll try to get this done ASAP.

**Comment by andvasp** [comment]
I looked the last commit and I noticed the error using the operator share in toaster.service.ts. I believe the code below would fix the problem.

// import 'rxjs/add/operator/share';
import {share} from 'rxjs/operators'

Instead of use .share() I use .pipe(share()) like shown below.
this.addToast = new Observable<Toast>((observer: any) => this._addToast = observer).pipe(share());

**Comment by sguiheux** [comment]
Still 1 error : 16 05 2018 12:27:55.224:ERROR [karma]: TypeError: Cannot read property '_tsInstances' of undefined

**Comment by fcalderon** [comment]
This is huge guys, thanks for doing this.

**Comment by Stabzs** [comment]
I've merged in 6.0.0 functionality.  Thank you for all of the hard work in this PR!  It was invaluable for finishing the implementation.

**Pull request closed without merging**